### PR TITLE
fix: updating links to be consistent with new specification pages

### PR DESCRIPTION
**Pull request by matebarabas**
## Description

updating links to be consistent with new specification pages

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (eriqua) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: Hey @matebarabas, we have a PR which is going to be merged soon ref #3803, moving `utilities` out of the avm folder.
After that all PRs updating CI scripts will need to pull upstream and work in the new utilities folder in the root

Comment (eriqua) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: Too late 😱 
But I think it's for the best. I would have suggested anyway to decouple static validation changes from the module ones. 

The reason is that changes to the CI (even if only documentational, like in this case, trigger all test deployments and we may want to avoid that and the consequent troubleshooting in this case.
Changes to the modules instead, since main.json is modified, qualify for a new patch version, hence we want them to trigger their respective pipelines.

To summarize, I'd suggest to please remove static validation changes from this PR and to open a dedicated one. The location of the utility folder is now the root of the repo. Its structure is the same as before.

Comment (matebarabas) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: I'll abandon this PR, and open 3 separate ones.

**Comment by matebarabas** [comment]
Closing this one, and opening 3 separate ones as requested/suggested by @eriqua.
